lib/active_model/serializer.rb

```diff
@@ -105,7 +105,7 @@ def self.root_name
       name.demodulize.underscore.sub(/_serializer$/, '') if name
     end
 
-    attr_accessor :object, :root, :meta, :meta_key, :scope
+    attr_accessor :object, :root, :meta, :meta_key, :links, :scope
```

Comment:
@joaomdmoura @kurko serializers are for the resource -> attributes mapping and adapters are for building the response object, right?

to ensure the `links` is an adapter-specific rendering option, it should really be passed to the adapter by adding `:links` to the list of ADAPTER_OPTION_KEYS at the top of [SerializableResource](https://github.com/rails-api/active_model_serializers/blob/b4949fbc2adb05f83e07ccca15abbf77cb3be43a/lib/active_model/serializable_resource.rb#L5).  That way, I think, we can keep the responsibilities of the the response in the adapter, and continue to limit the serialization of a resource to the serializer.

I wrote similarly in https://github.com/rails-api/active_model_serializers/pull/1041#discussion_r36708460 that I think `meta` should also be moved to the adapter, and is here mostly due to how the library has changed from 0.8 to 0.9 to 0.10 and the evolution of the json-api spec.

Your thoughts?  I think you can make this change mostly by deleting the code in the serializers that the adapter uses and just have the links passed directly to the adapter.